src/atomate2/qchem/sets/base.py

```diff
@@ -0,0 +1,486 @@
+        scan, van_der_waals, and plots. The value of each key is a dictionary
+        of key value pairs relevant to that section.
+        For example, to add a new variable to the rem section that sets
+        symmetry to false, use
+        overwrite_inputs = {"rem": {"symmetry": "false"}}
+        **Note that if something like basis is added to the rem dict it will overwrite
+        the default basis.**
+        **Note that supplying a van_der_waals section here will automatically modify
+        the PCM "radii" setting to "read".**
+        **Note that all keys must be given as strings, even when they are numbers!**
+
+    config_dict : dict
+        The config dictionary to use containing the base input set settings.
+    """
+
+    def __init__(
+        self,
+        job_type: str,
+        basis_set: str,
+        scf_algorithm: str,
+        dft_rung: int = 4,
+        pcm_dielectric: float | None = None,
+        smd_solvent: str | None = None,
+        custom_smd: str | None = None,
+        opt_variables: dict[str, list] | None = None,
+        scan_variables: dict[str, list] | None = None,
+        max_scf_cycles: int = 100,
+        geom_opt_max_cycles: int = 200,
+        plot_cubes: bool = False,
+        nbo_params: dict | None = None,
+        new_geom_opt: dict | None = None,
+        overwrite_inputs: dict | None = None,
+        vdw_mode: Literal["atomic", "sequential"] = "atomic",
+        rem_dict: dict | None = None,
+        opt_dict: dict | None = None,
+        pcm_dict: dict | None = None,
+        solv_dict: dict | None = None,
+        smx_dict: dict | None = None,
+        scan_dict: dict | None = None,
+        vdw_dict: dict | None = None,
+        plots_dict: dict | None = None,
+        nbo_dict: dict | None = None,
+        geom_opt_dict: dict | None = None,
+    ) -> None:
+        self.job_type = job_type
+        self.basis_set = basis_set
+        self.scf_algorithm = scf_algorithm
+        self.dft_rung = dft_rung
+        self.pcm_dielectric = pcm_dielectric
+        self.smd_solvent = smd_solvent
+        self.custom_smd = custom_smd
+        self.opt_variables = opt_variables
+        self.scan_variables = scan_variables
+        self.max_scf_cycles = max_scf_cycles
+        self.geom_opt_max_cycles = geom_opt_max_cycles
+        self.plot_cubes = plot_cubes
+        self.nbo_params = nbo_params
+        self.new_geom_opt = new_geom_opt
+        self.overwrite_inputs = overwrite_inputs
+        self.vdw_mode = vdw_mode
+
+        pcm_defaults = {
+            "heavypoints": "194",
+            "hpoints": "194",
+            "radii": "uff",
+            "theory": "cpcm",
+            "vdwscale": "1.1",
+        }
+
+        plots_defaults = {"grid_spacing": "0.05", "total_density": "0"}
+
+        opt_dict = {} if self.opt_variables is None else self.opt_variables
+
+        scan_dict = {} if self.scan_variables is None else self.scan_variables
```

Comment:
My advice would be to turn this into a `dataclass`. You can use `__post_init__` to do any validation, etc. As I see it the benefits are:
- It will allow you to reduce the amount of repetitive initialise code, e.g., `pcm_dict = pcm_dict or {}`.
- It will bring the input set classes inline with VASP, CP2K, FHI-aims.
- The subclasses for specific calculations will be simplified substantially.